main/query.py

```diff
@@ -762,3 +763,18 @@ def get_indexed_ref_by_query(
         raise RefNotFoundError(
             "Multiple references match query in given dataset",
             repr(Q))
+
+
+def estimate_count(table_name: str) -> int:
+    """Returns the estimated number of rows in a given table.
+
+    :param str table_name: table name
+    :rtype: int
+    """
+    with connection.cursor() as cursor:
+        cursor.execute(
+            "SELECT reltuples FROM pg_class WHERE relname = %s",
```

Comment:
NIT: Consider using f-strings.
`f"SELECT reltuples FROM pg_class WHERE relname = {table_name}"`

Reply:
Thanks for having a look.

> NIT: Consider using f-strings.
> `f"SELECT reltuples FROM pg_class WHERE relname = {table_name}"`

No, manually formatting values into the query string is considered a bad practice as it becomes vulnerable to SQL injection.

https://docs.djangoproject.com/en/5.0/topics/db/sql/#passing-parameters-into-raw says

> Do not use string formatting on raw queries or quote placeholders in your SQL strings!

with a couple examples below. I think your f-string example is similar to the first one.